### PR TITLE
Async version of the api.

### PR DESCRIPTION
**Pull request by Drummersbrother**
Hi! 
-----
#### Why did I make this use async?
I wanted to use this module for a discord bot, but I noticed it used requests, which was very slow in my usecase (making 20+ profile lookups as fast as possible), so I thought I'd try to create a version of the module that used `aiohttp` instead. 

#### Why might this be faster?
If you want to make many API requests quickly, the module would limit you to only one request at a time, which makes the whole process slow. But since this is an IO-bottlenecked process, we can speed it up by making multiple concurrent requests using the `aiohttp` module. The library would still be just as fast with single requests, but using async functions you can create many requests simultaneously.

#### Is it ready to merge?
This is not complete yet, it's still missing the ability to pass in your own `aiohttp.ClientSession`, but it's nearly there and I want to know if this feature is desired by the author @anthok at all.

#### Are there any new limitations or dependencies?
Some limitations of my implementation is it being restricted to `python >= 3.5`, and that it requires `aiohttp` and `async_timeout`. It also requires the `asyncio` module for the tests, but that's in the standard lib in 3.5+.

#### Why did you modify `setup.py`?
I don't know much about how `setup.py` is supposed to be structured, but I tried to insert the new dependencies where it looked like they should go.

#### Is your code tested?
TL;DR; No, but it should be possible to make it so.

I tried for a couple of hours to get the most simple async test to work, but I wasn't able to. This might be because I developed this on a chromebook, or it might be because of something else, I am still trying to figure out how to unittest asyncio code.

### CODE REVIEW

**Comment by anthok** [comment]
Thank you for this. I agree completely with you that asynchronous requesting should be added to remove the bottleneck. However, I know that the requests module itself has some functionality to do async. Let me dig around for a bit and I'll try to see what I can do.

**Comment by anthok** [comment]
@Drummersbrother what are your thoughts on moving to OWAPI and away from Lootbox.eu? We could also potentially support both APIs and add in the async wor.

**Comment by Drummersbrother** [comment]
@anthok since 
image
> Sorry guys but as of the moment we've discontinued support due to Studies + work, feel free to use other repo's or host yourself from our Source (available via Gitlabs: https://gitlab.com/SingularityIO/LootBoxAPI) Bearing in mind that it is not up to date in refference to Sombra+new seasons.

I guess the lootbox api will be down forever and won't be updated... 
So I think you/we/idk/whoever should move this to OWAPI instead, and I don't think there's much point in supporting lootbox, since it's discontinued (sadly).

Since I want to have access to some kind of overwatch python API lib, I'll probably use this pull request branch and create an async OWAPI api.

**Comment by anthok** [comment]
RIP lootbox.eu
I'm going to branch the repo in it's current state to something like "lootbox.eu"  and work on merging your stuff into master. I like this approach and think its the way to go.
